### PR TITLE
feat(database): support copy & paste of database

### PR DESCRIPTION
**Pull request by zqran**
https://user-images.githubusercontent.com/15389209/229132043-07846049-141c-4aad-b82c-9b82034902c4.mov

part of #629

### CODE REVIEW

**Review by doodlewind — APPROVED** [review]

File: packages/blocks/src/database-block/database-service.ts [hunk 1]
```diff
@@ -50,4 +54,70 @@ export class DatabaseBlockService extends BaseService<DatabaseBlockModel> {
...
+      }),
+    };
+  }
+
+  async onBlockPasted(
+    model: BlockModels['affine:database'],
+    props: Record<string, string[]>
+  ) {
+    // wait for the database's children to be added
+    queueMicrotask(() => {
```
Comment (doodlewind) [inline packages/blocks/src/database-block/database-service.ts:89]: It's best not to wait here

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -81,10 +81,11 @@ jobs:
 
       - name: Upload test results
         if: ${{ failure() }}
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v2
```
Comment (lawvs) [inline .github/workflows/test.yaml]: What's the problem with `upload-artifact@v3`?

Comment (zqran) [reply .github/workflows/test.yaml]: Just temporary debugging `CI` tests

**Comment by zqran** [comment]
The failed test may be fixed by: #1936

**Pull request merged**